src/components/calcite-inline-editable/calcite-inline-editable.tsx

```diff
@@ -0,0 +1,170 @@
+import { Component, Element, h, Host, Listen, Prop, VNode } from "@stencil/core";
+
+@Component({
+  tag: "calcite-inline-editable",
+  shadow: true,
+  styleUrl: "calcite-inline-editable.scss"
+})
+export class CalciteInlineEditable {
+  //--------------------------------------------------------------------------
+  //
+  //  Element
+  //
+  //--------------------------------------------------------------------------
+
+  @Element() el!: HTMLCalciteInlineEditableElement;
+
+  //--------------------------------------------------------------------------
+  //
+  //  Props
+  //
+  //--------------------------------------------------------------------------
+
+  @Prop({ mutable: true, reflect: true }) editingEnabled = false;
+
+  @Prop({ mutable: true, reflect: true }) loading = false;
+
+  @Prop() onConfirmChanges: () => Promise<any>;
```

Comment:
This should probably be an `@Event` instead of a prop.  Stencil is warning about this in the console
![image](https://user-images.githubusercontent.com/821864/97480156-4738ad80-1910-11eb-9b58-ad0551462a37.png)

Reply:
I looked into how to silence this warning, but couldn't figure it out. Seems to be driven by the `onXYZ` naming convention. This prop is the result of feedback from my previous PR which was purely controlled and all event driven. The downside to that approach was that it required the consumer to manage all the other UI states (i.e. editing state, loading). That was a bit cumbersome and this approach makes working with the component quite a bit easier. There is precedent for passing callbacks to components (see `beforeClose`). However, if a consumer absolutely needs to manage all the other states and wants to opt out of using the callback, there is now also now a way to do that.